src/legacy/core_plugins/tile_map/public/tile_map_type.js

```diff
@@ -113,7 +113,7 @@ export function createTileMapTypeDefinition(dependencies) {
         ],
         tmsLayers: [],
       },
-      optionsTemplate: withInjectedDependencies(TileMapOptions, { serviceSettings }),
+      optionsTemplate: (props) => <TileMapOptions {...props} serviceSettings={serviceSettings} />,
```

Comment:
:information_source: I inlined this, since I think that's better readable than excluding it in an additional utility (also we save another file :D)